### DISCUSSION TITLE
[Android][Huawei] Failed to add ruleset (PublicKeyToken=Null)

### DISCUSSION THREAD

**Discussion by himekode** [ppy/osu#18860] [Q&A] [answered]
### Type

Game behaviour

### Bug description

Hi. I am kind of new in Github, so I hope I am doing this right.

I wanted to load/add Hishigata and Touhosu on my phone, but all it says after I launch the app is that it "failed to add ruleset", along with the rulesets' information."failed to add ruleset", along with the rulesets' information.

I have searched the Net, and I have not got any resolution for this.

Please help me resolve this. Thank you!

### Screenshots or videos

![Screenshot_20220627_104833_sh ppy osulazer](https://user-images.githubusercontent.com/108086202/175851464-7d61fea3-bc83-42a8-8b8c-7b48ec8b029f.jpg)

### Version

2022.621.0

### Logs

[runtime.log](https://github.com/ppy/osu/files/8988341/runtime.log)
[performance.log](https://github.com/ppy/osu/files/8988342/performance.log)
[network.log](https://github.com/ppy/osu/files/8988343/network.log)
[database.log](https://github.com/ppy/osu/files/8988344/database.log)

**Comment by LumpBloom7** [comment, marked as answer]
Custom ruleset dlls can't be installed like that on mobile.

Rulesets need to be included during a mobile lazer build, in order for them to be properly installed. This could be done via self/unofficial builds.

**Reply by himekode** [reply]
Thanks for the answer.

I am a newb for this, so can you recommend me some builds that are able to add rulesets? Thanks.

**Reply by peppy** [reply]
Please use the PC releases for now.
